tofhir-engine/src/main/scala/io/tofhir/engine/data/write/FhirRepositoryWriter.scala

```diff
@@ -145,76 +145,60 @@ class FhirRepositoryWriter(sinkSettings: FhirRepositorySinkSettings, runningJobR
-                        expression = Some(Serialization.write(validationErrorsOfMappings.getOrElse(index, None)))
-                      )
-                    )
+          logger.error(msg, fce)
+          // find the validation errors of mappings
+          val validationErrorsOfMappings = groupOutcomeIssuesByEntryIndex(fce.serverResponse.get.outcomeIssues)
+          mappingResults
+            .zipWithIndex.map {
+            case (element, index) =>
+              element._2.copy(
+                error = Some(
+                  FhirMappingError(
+                    code = FhirMappingErrorCodes.INVALID_RESOURCE,
+                    description = "Resource is not a valid FHIR resource or conforming to the indicated profiles!",
+                    expression = Some(Serialization.write(validationErrorsOfMappings.getOrElse(index, None)))
                   )
-              }.foreach(failedResult => problemsAccumulator.add(failedResult))
-            None
-          }
+                )
+              )
+          }.foreach(failedResult => problemsAccumulator.add(failedResult))
+          None
         } else {
           val msg = s"FHIR repository at url ${sinkSettings.fhirRepoUrl} returned an unidentified error while writing the resources!"
-          if (sinkSettings.errorHandling.isEmpty || sinkSettings.errorHandling.get == ErrorHandlingType.HALT) {
-            logger.error(msg, fce)
-            throw FhirMappingException(msg, fce)
-          } else {
-            mappingResults
-              .map(_._2)
-              .map(mr =>
-                mr.copy(error = Some(FhirMappingError( //Set the error
-                  code = FhirMappingErrorCodes.SERVICE_PROBLEM,
-                  description = msg
-                )))
-              ).foreach(failedResult => problemsAccumulator.add(failedResult))
-            None
-          }
-        }
-      case tout: TimeoutException =>
-        val msg = s"FHIR repository at url ${sinkSettings.fhirRepoUrl} timeout for batch interaction while writing the resources!"
-        if (sinkSettings.errorHandling.isEmpty || sinkSettings.errorHandling.get == ErrorHandlingType.HALT) {
-          logger.error(msg, tout)
-          throw FhirMappingException(msg, tout)
-        } else {
+          logger.error(msg, fce)
           mappingResults
             .map(_._2)
             .map(mr =>
               mr.copy(error = Some(FhirMappingError( //Set the error
-                code = FhirMappingErrorCodes.FHIR_API_TIMEOUT,
+                code = FhirMappingErrorCodes.SERVICE_PROBLEM,
                 description = msg
               )))
             ).foreach(failedResult => problemsAccumulator.add(failedResult))
           None
         }
+      case tout: TimeoutException =>
+        val msg = s"FHIR repository at url ${sinkSettings.fhirRepoUrl} timeout for batch interaction while writing the resources!"
+        logger.error(msg, tout)
```

Comment:
We did not have this logging in the previous version. This may lead duplicate logs since we pass the error to problemsAccumulator ?